core/src/main/java/tc/oc/pgm/commands/ListCommands.java

```diff
@@ -0,0 +1,112 @@
+package tc.oc.pgm.commands;
+
+import app.ashcon.intake.Command;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.command.CommandSender;
+import tc.oc.pgm.api.Permissions;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.VanishManager;
+import tc.oc.pgm.teams.TeamMatchModule;
+import tc.oc.pgm.util.component.Component;
+import tc.oc.pgm.util.component.Components;
+import tc.oc.pgm.util.component.types.PersonalizedText;
+import tc.oc.pgm.util.component.types.PersonalizedTranslatable;
+import tc.oc.pgm.util.named.NameStyle;
+
+public class ListCommands {
+
+  private final VanishManager vanish;
+
+  public ListCommands(VanishManager vanish) {
+    this.vanish = vanish;
+  }
+
+  @Command(
+      aliases = {"list", "who", "online", "ls"},
+      desc = "View a list of online players")
+  public void list(CommandSender sender, MatchPlayer player, Match match) {
+    TeamMatchModule tmm = match.getModule(TeamMatchModule.class);
+    if (tmm != null) {
+      sender.sendMessage(
+          new PersonalizedTranslatable("commands.users.teams")
+              .getPersonalizedText()
+              .color(ChatColor.GRAY));
+      tmm.getTeams()
+          .forEach(
+              team ->
+                  sendTeamInfo(
+                      sender, team.getComponentName(), team.getPlayers(), team.getMaxPlayers()));
+    } else {
+      sendTeamInfo(
+          sender,
+          new PersonalizedTranslatable("commands.users.participants"),
```

Comment:
It would be great if we can start using `TextComponent` and `TranslatableComponent` with the new text library. Soon I won't be accepting any more changes with the old text. Or else we'll never migrate.

Reply:
I was wondering about this, as I seem to have run into a few issues. First using `MatchPlayer#showHotbar` did not actually show the hotbar. After searching the repo, I was unable to find any calls to the new hotbar, so not sure if that’s been tested. 

Also in `ListCommands` I’m utilizing `Components#join` to neatly format the player names. Since that’s still using the old component system, what should be done there? I can migrate the method if you like, however there are quite a few calls to that method from `ModerationCommands`.

 I’m planning on a PR soon that will have some changes to `ModerationCommands`, maybe it would be more appropriate to migrate all the moderations commands to the new text library then? 

I did however use `TranslatableComponent` for the main vanish commands. 👍

Reply:
Great, if you have some blockers don’t worry about it for now.